tests/test_modeling_flax_vision_encoder_decoder.py

```diff
@@ -0,0 +1,513 @@
+# coding=utf-8
+# Copyright 2021 HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import tempfile
+import unittest
+
+import numpy as np
+
+from transformers import is_flax_available, is_torch_available, is_vision_available
+from transformers.testing_utils import is_pt_flax_cross_test, require_flax, require_vision, slow, torch_device
+
+from .test_modeling_flax_common import floats_tensor, ids_tensor
+from .test_modeling_flax_gpt2 import FlaxGPT2ModelTester
+from .test_modeling_flax_vit import FlaxViTModelTester
+
+
+if is_flax_available():
+    from transformers import (
+        AutoTokenizer,
+        FlaxGPT2LMHeadModel,
+        FlaxVisionEncoderDecoderModel,
+        FlaxViTModel,
+        VisionEncoderDecoderConfig,
+    )
+    from transformers.modeling_flax_pytorch_utils import (
+        convert_pytorch_state_dict_to_flax,
+        load_flax_weights_in_pytorch_model,
+    )
+
+if is_torch_available():
+    import torch
+
+    from transformers import VisionEncoderDecoderModel
+
+if is_vision_available():
+    from PIL import Image
+
+    from transformers import ViTFeatureExtractor
+
+
+@require_flax
+class FlaxEncoderDecoderMixin:
+    def get_encoder_decoder_model(self, config, decoder_config):
+        raise NotImplementedError
+
+    def prepare_config_and_inputs(self):
+        raise NotImplementedError
+
+    def get_pretrained_model(self):
+        raise NotImplementedError
```

Comment:
are these not required? Maybe we could just remove these then.

Reply:
Actually, I copied this from `test_modeling_encoder_decoder.py` when I worked on other (text/vision) encoder-decoder models. I don't check the reason why these lines are there, and if it applies to Flax's version. I would prefer to remove them (at the same time if possible) after some investigations in another PR, and maybe having others' opinions.

Reply:
I don't think these apply to flax version, so we could remove these.